### PR TITLE
feat: add retry informational headers

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Internal issue in JS-1816

*Description of changes:*
add retry informational headers

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/middleware-retry/package.json [hunk 1]
```diff
@@ -14,12 +14,13 @@
   },
   "license": "Apache-2.0",
   "dependencies": {
+    "@aws-sdk/protocol-http": "1.0.0-gamma.1",
     "@aws-sdk/service-error-classification": "1.0.0-gamma.1",
     "@aws-sdk/types": "1.0.0-gamma.1",
-    "tslib": "^1.8.0"
+    "tslib": "^1.8.0",
+    "uuid": "^8.0.0"
```
Comment (AllanFly120) [inline packages/middleware-retry/package.json:21]: Just fyi, uuid doesn't work in RN directly. See here: https://github.com/uuidjs/uuid#react-native

**Comment by trivikr** [comment]
Created a PR with upstream https://github.com/aws/aws-sdk-js-v3/pull/1268
Reason: GitHub Action not able to fetch credentials on PR created from fork https://github.com/aws/aws-sdk-js-v3/issues/1267

**Comment by trivikr** [comment]
Reviving this PR as running CI using GitHub action was reverted in #1278

File: packages/middleware-retry/package.json (hunk 1 above)
Comment (trivikr) [reply packages/middleware-retry/package.json:21]: Refs: [#1216/commits/18945e17b663ec3e0b52187521017709576273ad](https://github.com/aws/aws-sdk-js-v3/pull/1216/commits/18945e17b663ec3e0b52187521017709576273ad)

Comment (trivikr) [reply packages/middleware-retry/package.json:22]: Updated in [commit](https://github.com/aws/aws-sdk-js-v3/pull/1266/commits/c29719146b02577475624ca0e810decde69d46fd)

**Pull request merged**